### PR TITLE
Add SensorShapeValidator unit

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Add unit tests for SensorShapeValidator, fix potential errors.

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Sensor/SensorShapeValidator.cs [hunk 1]
```diff
@@ -27,12 +27,12 @@ public void ValidateSensors(List<ISensor> sensors)
                 // Check for compatibility with the other Agents' Sensors
                 // TODO make sure this only checks once per agent
                 Debug.Assert(m_SensorShapes.Count == sensors.Count, $"Number of Sensors must match. {m_SensorShapes.Count} != {sensors.Count}");
-                for (var i = 0; i < m_SensorShapes.Count; i++)
+                for (var i = 0; i < Mathf.Min(m_SensorShapes.Count, sensors.Count); i++)
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensor/SensorShapeValidator.cs]: If these differed before, we'd get the assert, but would potentially continue and iterate through the lists, so accessing `sensors[i]` would read past the end of the list.

File: com.unity.ml-agents/Tests/Editor/Sensor/SensorShapeValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+using UnityEngine.TestTools;
+
+namespace MLAgents.Tests
+{
+    public class NoopSensor : ISensor
+    {
+        string m_Name = "NoopSensor";
+        int[] m_Shape;
+
+        public NoopSensor(int dim1)
```
Comment (chriselion) [inline com.unity.ml-agents/Tests/Editor/Sensor/SensorShapeValidatorTests.cs]: Just some convenience methods for making sensors of a certain size.

File: com.unity.ml-agents/Tests/Editor/Sensor/SensorShapeValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
+using System.Collections.Generic;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace MLAgents.Tests
+{
+    public class NoopSensor : ISensor
```
Comment (vincentpierre) [inline com.unity.ml-agents/Tests/Editor/Sensor/SensorShapeValidatorTests.cs]: What is a Noop?

Comment (chriselion) [reply com.unity.ml-agents/Tests/Editor/Sensor/SensorShapeValidatorTests.cs]: https://en.wikipedia.org/wiki/NOP_(code)

Comment (chriselion) [reply com.unity.ml-agents/Tests/Editor/Sensor/SensorShapeValidatorTests.cs]: I can rename to DummySensor or something clearer

Comment (surfnerd) [reply com.unity.ml-agents/Tests/Editor/Sensor/SensorShapeValidatorTests.cs]: NoOpSensor?

**Pull request merged**